src/fancyline/tty/vt100.cr

```diff
@@ -1,5 +1,25 @@
 class Fancyline
   class Tty
+    lib LibC
+      struct Winsize
+        ws_row    : ::LibC::UShort # rows, in characters
+        ws_col    : ::LibC::UShort # columns, in characters
+        ws_xpixel : ::LibC::UShort # horizontal size, pixels
+        ws_ypixel : ::LibC::UShort # vertical size, pixels
+      end
+
+      {% if flag?(:x86_64) && flag?(:darwin) %}
+        IOC_OUT      = 0x40000000
+        IOCPARM_MASK =     0x1fff
+        TIOCGWINSZ   = IOC_OUT | ((sizeof(Winsize) & IOCPARM_MASK) << 16) | (('t'.ord) << 8) | 104
+      {% else %}
+        {% puts "Warning: Tty::Vt100#winsize is not supported on your platform." %}
+      {% end %}      
+
+      @[Raises]
```

Comment:
`@[Raises]` is to mark C functions that actually throw an exception, which is not done through returning something.  As `ioctl()` doesn't raise (but returns an error code), this attribute doesn't do much.

Reply:
Ok, I thought it threw exceptions, but yeah, error codes, the world of C is pretty alien to me (unfortunately [as I really appreciate the language]).